tests/ignite/contrib/metrics/test_roc_curve.py

```diff
@@ -121,3 +132,26 @@ def test_check_compute_fn():
 
     em = RocCurve(check_compute_fn=False)
     em.update(output)
+
+
+def test_distrib_integration(distributed):
+    rank = idist.get_rank()
+    torch.manual_seed(41 + rank)
+    n_batches, batch_size = 5, 10
+    y = torch.randint(0, 2, size=(n_batches * batch_size,))
+    y_pred = torch.rand((n_batches * batch_size,))
+
+    device = "cpu" if idist.device().type == "xla" else idist.device()
+    metric = RocCurve(device=device)
+    for i in range(n_batches):
+        metric.update((y_pred[i * batch_size : (i + 1) * batch_size], y[i * batch_size : (i + 1) * batch_size]))
```

Comment:
Can you replace this using `Engine` and a proper engine integration. There are similar tests for other contrib metrics to inspire from.